app/services/vot/parser.rb

```diff
@@ -4,8 +4,9 @@ module Vot
   class Parser
     class ParseException < StandardError; end
 
-    Result = Data.define(
+    class Result < Data.define(
```

Comment:
same `Data.define` comment as above

Reply:
I don't think the same case can be made here given how complex it is already.

Reply:
This probably should have an initializer as well. Thoughts?

Reply:
if it's being turned into a class, then yes i think we should have a real initializer rather than using Data.define. my personal preference would be to leave it alone for this PR, but the complexity comment is valid, so up to you whether to pull out the initializer or revert it to a class-like object!

Reply:
ok, sorry thinking through this a little more:

as mentioned, the [Data class was explicitly defined](https://docs.ruby-lang.org/en/master/Data.html) for use on class-like objects. it was not designed to be used in a class inheritance way as is being used here. if we wanted to fully convert the class, that would require removing this and updating all the references from using `result.with`. 

i *do not* think we should do that as part of this PR (which is complicated enough already), so i'd like to more strongly encourage reverting this to `Result = Data.define` for the purposes of this PR, and adding a ticket to evaluate this more fully as part of next steps. i don't think converting this into a full class has any particularly strong benefits for the purpose of this change.